streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableGlobalKTableJoin.java

```diff
@@ -74,26 +79,29 @@ public void enableSendingOldValues() {
     }
 
 
-    private class KTableGlobalKTableJoinProcessor extends AbstractProcessor<K1, Change<V1>> {
+    private class KTableGlobalKTableJoinProcessor extends AbstractProcessor<K, Change<V>> {
 
-        private final KTableValueGetter<K2, V2> valueGetter;
+        private final KTableValueGetter<GK, GV> valueGetter;
+        private KeyValueStore<K, R> joinResultStore;
 
-        KTableGlobalKTableJoinProcessor(final KTableValueGetter<K2, V2> valueGetter) {
+        KTableGlobalKTableJoinProcessor(final KTableValueGetter<GK, GV> valueGetter) {
             this.valueGetter = valueGetter;
         }
 
         @SuppressWarnings("unchecked")
         @Override
-        public void init(ProcessorContext context) {
+        public void init(final ProcessorContext context) {
             super.init(context);
             valueGetter.init(context);
+            joinResultStore = (KeyValueStore<K, R>) context.getStateStore(joinResultStoreName);
+            ((CachedStateStore) joinResultStore).setFlushListener(new ForwardingCacheFlushListener<K, V>(context, sendOldValues));
         }
 
         /**
          * @throws StreamsException if key is null
          */
         @Override
-        public void process(final K1 key, final Change<V1> change) {
+        public void process(final K key, final Change<V> change) {
```

Comment:
I'm not sure i follow. Not immediately clear to me how `KTableKTableXXJoin` can't expect a `Change<>`. That is what is will be sent from the previous `KTableXXXProcessor`